### PR TITLE
Feat: Add light sensor component for veml7700 light sensor

### LINKED ISSUE #32: Light sensor - VEML7700 Component
Implement as firmware-idf component from esp-idf-lib

### PR DESCRIPTION
**Pull request by Ricovl**
Closes #32

### CODE REVIEW

File: components/light_sensor/include/light_sensor.hpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#define LIGHT_SENSOR_HPP
+
+#include <i2cdev.h>
+#include <veml7700.h>
+
+#include "component.hpp"
+
+namespace light_sensor {
+
+class light_sensor : public component {
```
Comment (SquaredPotato) [inline components/light_sensor/include/light_sensor.hpp]: indentation in this class is wacky, please use the .clang_format

File: components/light_sensor/src/light_sensor.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    m_conf.shutdown = 0;
+    err = veml7700_set_config(&m_dev, &m_conf);
+    if (err) {
+        RETURN_ON_ERR_MSG(err, "Failed to set light sensor config: ");
+    }
+
+    m_status = RUNNING;
+	return Ok(INITIALIZING);
```
Comment (SquaredPotato) [inline components/light_sensor/src/light_sensor.cpp]: extra indent?

File: components/light_sensor/include/light_sensor.hpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	light_sensor() {};
+	~light_sensor() = default;
+
+	/* Component override functions */
+	etl::string<50> get_tag() override { return TAG; };
+	res get_status() override;
+	res initialize() override;
+	res stop() override;
+
+    Result<uint32_t, etl::string<128>> read_light_level();
```
Comment (SquaredPotato) [inline components/light_sensor/include/light_sensor.hpp]: We should add a type in the SDK for error strings, I'll add an issue

**Review by SquaredPotato — CHANGES_REQUESTED** [review]

**Review by SquaredPotato — APPROVED** [review]

File: components/light_sensor/src/light_sensor.cpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        m_conf.shutdown = 0;
+        err             = veml7700_set_config(&m_dev, &m_conf);
+        if (err) { RETURN_ON_ERR_MSG(err, "Failed to set light sensor config: "); }
+
+        m_status = RUNNING;
+        return Ok(INITIALIZING);
+    }
+
+    res light_sensor::stop() {
+        m_conf.shutdown = 1;
```
Comment (abrondijk) [inline components/light_sensor/src/light_sensor.cpp:40]: What does this do?

**Review by abrondijk — APPROVED** [review]

**Pull request merged**